sunsynk/sensor.py

```diff
@@ -42,11 +42,12 @@ class Sensor:
     name: str = attr.field()
     unit: str = attr.field(default="")
     factor: float = attr.field(default=1)
-    value: Union[float, int, str, None] = None
     # func: Union[
     #     None, Callable[[Tuple[int, ...]], str], Callable[[float], Any]
     # ] = attr.field(default=None)
     reg_value: Tuple[int, ...] = attr.field(init=False, factory=tuple)
+    on_change: Callable = attr.field(default=None)
```

Comment:
An `on_change` callable was introduced here, but change detection could be implemented in the add-on itself and the sunsynk library left clean.

Reply:
Yes, that is exactly why I invoked the `on_change` callback (if set) when setting `value`. The `value` does get set when reading a new value (for example, me walking to the inverter and updating the value there), so that case is covered too.

I initially had the callback invoked inside the `update_value()` method, however that is overridden in subclasses and the callback ended up not being invoked. That is the exact reason why I invoke the callback in the setter of `value`.